packages/terra-clinical-result/src/ClinicalResultBloodPressure.jsx

```diff
@@ -120,144 +120,144 @@ const ClinicalResultBloodPressure = (props) => {
+  const hasUnverifiedIcon = (systolic && systolic.isUnverified) || (diastolic && diastolic.isUnverified);
+  const datetimeDisplayElement = createDatetimeDisplays(compareDatetimeDisplays);
+  const conceptDisplayElement = createConceptDisplays(compareConceptDisplays);
 
-    const decoratedResultDisplay = [];
-    if (systolic || diastolic) {
-      if (!systolic) {
-        decoratedResultDisplay.push(<ResultError key={`Error-Systolic-${id}`} />);
-      } else if (noDataSystolic) {
-        decoratedResultDisplay.push(<NoData key={`NoData-Systolic-${id}`} />);
-      } else if (systolic) {
-        const systolicDisplay = (
-          <ConditionalWrapper
-            key={`del-Systolic-${systolic.eventId}`}
-            condition={compareStatusIsInError.systolic}
-            wrapper={children => <del>{children}</del>}
-          >
-            <Observation
-              key={`Observation-Systolic-${systolic.eventId}`}
-              eventId={systolic.eventId}
-              result={systolic.result}
-              interpretation={!compareStatusIsInError.systolic ? systolic.interpretation : null}
-              isUnverified={systolic.isUnverified}
-              hideUnit={hideUnit || ((compareUnits.systolic === compareUnits.diastolic) && !compareStatusIsInError.systolic)}
-            />
-          </ConditionalWrapper>
-        );
-        decoratedResultDisplay.push(systolicDisplay);
-      }
-      decoratedResultDisplay.push(<span key={`Observation-Separator-${(systolic) ? systolic.eventId : diastolic.eventId}`} className={cx('result-display-separator')}>/</span>);
-      if (!diastolic) {
-        decoratedResultDisplay.push(<ResultError key={`Error-Diastolic-${id}`} />);
-      } else if (noDataDiastolic) {
-        decoratedResultDisplay.push(<NoData key={`NoData-Diastolic-${id}`} />);
-      } else if (diastolic) {
-        const diastolicDisplay = (
-          <ConditionalWrapper
-            key={`del-Diastolic-${diastolic.eventId}`}
-            condition={compareStatusIsInError.diastolic}
-            wrapper={children => <del>{children}</del>}
-          >
-            <Observation
-              key={`Observation-Diastolic-${diastolic.eventId}`}
-              eventId={diastolic.eventId}
-              result={diastolic.result}
-              interpretation={!compareStatusIsInError.diastolic ? diastolic.interpretation : null}
-              isUnverified={diastolic.isUnverified}
-              hideUnit={hideUnit}
-            />
-          </ConditionalWrapper>
-        );
-        decoratedResultDisplay.push(diastolicDisplay);
-      }
+  const decoratedResultDisplay = [];
+  if (systolic || diastolic) {
+    if (!systolic) {
+      decoratedResultDisplay.push(<ResultError key={`Error-Systolic-${id}`} />);
+    } else if (noDataSystolic) {
+      decoratedResultDisplay.push(<NoData key={`NoData-Systolic-${id}`} />);
+    } else if (systolic) {
```

Comment:
this block is massive :( would it make more sense to create helpers based on systolic and diastolic? we can make the code easier to read. something similar to below:

```js
/* helpers. first two could be further reduced by another helper */
const createSystolicDisplay(systolic) {
  if (systolic) { 
    // return wrapper
  } else if (systolic.noDataSystolic) { 
    // return no data
  } else { 
    // return error 
  }
}

const createSystolicDisplay(diastolic) {
  if (diastolic) { 
    // return wrapper
  } else if (diastolic.noDataDiastolic) { 
    // return no data
  } else { 
    // return error 
  }

const createResultDisplay = (systolic, diastolic) => {
  if (!systolic && !diastolic) {
    return;
  } else {
    const resultDisplay = [];
    resultDisplay.push(createSystolicDisplay(systolic);
    resultDisplay.push(separator);
    resultDisplay.push(createDiastolicDisplay(systolic);
    return resultDisplay;
  }

...
/* use helper */
const resultDisplay = createResultDisplay(systolic, diastolic);
```

what are your thoughts?

Reply:
Split it up, but didn't make the createResultDisplay function, because we have a few others things that get set under that check.
https://github.com/cerner/terra-clinical/pull/648/commits/5611e410a5b4c6389abf03f7d50fbb975c73143e#diff-13926e711e02256769a4612c0d911268R106